### PR TITLE
feat: throttle in source

### PR DESCRIPTION
**Pull request by tabVersion**
Signed-off-by: tabVersion <tabvision@bupt.icu>

I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

as title, rate limit in source and allow config change to adjust

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer the issue)
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: src/stream/src/executor/source/source_executor.rs [hunk 1]
```diff
@@ -260,6 +262,9 @@ impl<S: StateStore> SourceExecutor<S> {
             }
         }
 
+        let mut max_message_per_sec = 1024;
+        let rate_limit_interval = Duration::from_millis(1000);
```
Comment (xx01cyx) [inline src/stream/src/executor/source/source_executor.rs:266]: Are these fair initial values?

**Review by xx01cyx — COMMENTED** [review]
Generally LGTM. Any test on it?

**Comment by BugenZhao** [comment]
Would you mind telling me the motivation or design of this PR?

**Comment by KeXiangWang** [comment]
> Would you mind telling me the motivation or design of this PR?

I need this PR to do some POC and I ask Bohan for help, not intend to merge into Main branch.

File: src/stream/src/executor/source/source_executor.rs (hunk 1 above)
Comment (tabVersion) [reply src/stream/src/executor/source/source_executor.rs:266]: just placeholder, kexiang will look into it

**Pull request closed without merging**